plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/WidgetHandler.java

```diff
@@ -74,6 +74,29 @@ public boolean isEnabled(Widget widget) {
 		}
 		return ret;
 	}
+	
+	/**
+	 * Finds out whether specified {@link Widget} is disposed or not.
+	 * 
+	 * @param widget widget to handle
+	 * @return true if widget is disposed, false otherwise
+	 */
+	public boolean isDisposed(Widget widget) {
+		boolean ret = true;
+		Object o = null;
+		try {
```

Comment:
Why use reflection? Why not just directly call widget.isDisposed() method in UIThread?